geoh5py/shared/merging/base.py

```diff
@@ -91,12 +91,15 @@ def merge_data(
                         if association == "VERTEX"
                         else out_entity.n_cells
                     )
+
                     data_dict[label] = out_entity.add_data(
                         {
                             data.name: {
                                 "values": np.ones(shape) * data.nan_value,
                                 "association": association,
-                                "entity_type": data.entity_type,
+                                "entity_type": data.entity_type.copy(
+                                    out_entity.workspace
```

Comment:
Any chance we can move this to the `object_base.validate_data_type`. That way we can catch more possible issues with people adding data and sharing entity_type. Just need to add a little logic to check if the workspace differ.

Reply:
We can't check `entity_type.workspace` 
If we try to access it with the property, we'll raise an assertion error (see my previous comment).
So: 
- or we remove this assertion error in `entity_type.workspace`
- or we access with `getattr()`

Could be another issue, as this option is not concerning the merging but a more global issue?

Reply:
Don't see how that's possible. Maybe we should talk